backend/services/interfaces/testSessionService.ts

```diff
@@ -34,18 +29,18 @@ export interface TestSessionResponseDTO {
   /** the unique identifier for the test session */
   id: string;
   /** the corresponding test from the Test collection */
-  test: TestResponseDTO;
+  test: string;
   /** the teacher administering the test from the User collection */
-  teacher: UserDTO;
+  teacher: string;
   /** the school that's administering the test from the School collection */
-  school: SchoolResponseDTO;
+  school: string;
   /** the class taking the test session */
-  class: ClassResponseDTO;
+  class: string;
   /**
    * the result of the test session
    * there should be one entry here per student
    */
-  results: ResultResponseDTO[];
+  results?: ResultResponseDTO[];
```

Comment:
shouldn't the response always have a results field? (but could be `[]`)

Reply:
My understanding is that it could be missing from Mongo objects. Previously, we manually filled in this field before returning it from the service, but I think it's more correct to have GraphQL handle this sort of defaulting (hence the fact that the TypeScript type makes the field optional; the GraphQL type still has it as a required field). But I'm open to alternatives depending on your thoughts here.